src/controller/data_model/BUILD.gn

```diff
@@ -22,7 +22,9 @@ chip_data_model("data_model") {
   zap_pregenerated_dir =
       "${chip_root}/zzz_generated/controller-clusters/zap-generated"
 
-  use_tests_apis = true
+  # On Android, we don't really need the tests APIs, and the compiler has
+  # trouble compiling some of those files anyway.
+  use_tests_apis = current_os != "android"
```

Comment:
consider setting it default to false and have chip-tool on linux and darwin explicitly enable it.

I believe most platforms that want to compile a controller data model have no intention of running unit tests.

Reply:
Does the python controller want it?

I believe the Darwin framework wants it for its tests...

Reply:
Will add TODO to figure this out and have all the consumers that need it opt in explicitly.